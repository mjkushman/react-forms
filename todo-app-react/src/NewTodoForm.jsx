import Todo from "./Todo";
import { useState } from "react";


const NewTodoForm = ({addTodo}) => {
    const INITIAL_STATE = {
        task: ""
    }
    const [formData, setFormData] = useState(INITIAL_STATE)

    const handleSubmit = (e) => {
        e.preventDefault()
        addTodo(formData)
        setFormData(INITIAL_STATE)
    }

    const handleChange = (e) => {
        const {name,value} = e.target;
        setFormData(formData => ({
            ...formData,
            [name]:value
        }));
    }

    return (
        
        <form onSubmit={handleSubmit}>
            <label htmlFor="task"> Task </label>
            <input 
            name="task"
            id="task"
            type="text"
            value={formData.task}
            onChange={handleChange}
            />

        <button>Add</button>

        </form>
        
    )
}

export default NewTodoForm