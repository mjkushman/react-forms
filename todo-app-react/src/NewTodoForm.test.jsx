import React from 'react'
import { render } from '@testing-library/react';
import {describe, test, expect} from 'vitest';
import NewTodoForm from './NewTodoForm.jsx'


// Smoke test
test('Renders without going up in smoke', ()=> {
  render(<NewTodoForm />)
})

// snapshot test
test("snapshot match", ()=> {
  const {asFragment} = render(<NewTodoForm/>);

  expect(asFragment()).toMatchSnapshot();
})