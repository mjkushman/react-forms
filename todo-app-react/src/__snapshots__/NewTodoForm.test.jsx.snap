// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot match 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
       Task 
    </label>
    <input
      id="task"
      name="task"
      type="text"
      value=""
    />
    <button>
      Add
    </button>
  </form>
</DocumentFragment>
`;
